lib/pbench/test/unit/server/query_apis/test_datasets_list.py

```diff
@@ -136,3 +136,127 @@ def test_query(
+            "_shards": {"total": 5, "successful": 5, "skipped": 0, "failed": 0},
+            "hits": {
+                "total": {"value": 2, "relation": "eq"},
+                "max_score": None,
+                "hits": [
+                    {
+                        "_index": "drb.v6.run-data.2020-04",
+                        "_type": "_doc",
+                        "_id": "12fb1e952fd826727810868c9327254f",
+                        "_score": None,
+                        "_source": {
+                            "authorization": {"access": "private", "user": "unknown"},
+                            "@metadata": {"controller_dir": "dhcp31-187.example.com"},
+                            "run": {
+                                "controller": "node",
+                                "name": "drb",
+                                "start": "2020-04-29T12:49:13.560620",
+                                "end": "2020-04-29T13:30:04.918704",
+                                "id": "12fb1e952fd826727810868c9327254f",
+                                "config": "rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus",
+                            },
+                        },
+                        "sort": [1588167004918],
+                    },
+                    {
+                        "_index": "drb.v6.run-data.2020-04",
+                        "_type": "_doc",
+                        "_id": "12fb1e952fd826727810868c9327254f",
+                        "_score": None,
+                        "_source": {
+                            "authorization": {"access": "private", "user": "unknown"},
+                            "@metadata": {"controller_dir": "dhcp31-187.example.com"},
+                            "run": {
+                                "controller": "node",
+                                "name": "test",
+                                "start": "2020-04-29T12:49:13.560620",
+                                "end": "2020-04-29T13:30:04.918704",
+                                "id": "12fb1e952fd826727810868c9327254f",
+                                "config": "rhel8_kvm_perf43_preallocfull_nvme_run4_iothread_isolcpus",
+                            },
+                        },
+                        "sort": [1588167004918],
+                    },
+                ],
+            },
+        }
+
+        index = self.build_index(
+            server_config, self.date_range(self.payload["start"], self.payload["end"])
+        )
+
+        response = query_api(
+            "/datasets/list",
+            "/_search?ignore_unavailable=true",
+            payload,
+            index,
+            HTTPStatus.OK,
+            headers={"authorization": f"Bearer {pbench_token}"},
+            json=response_payload,
+        )
+        assert response.status_code == HTTPStatus.OK
+        res_json = response.json
+        assert isinstance(res_json, dict)
+        assert len(res_json.keys()) == 1
```

Comment:
You don't need to explicitly invoke `keys()` -- `len()` when applied to a `dict` returns the number of keys.  (Actually, I think a `dict` referenced in a "list context" yields its keys...but it's the same difference....)

Reply:
That's true; this was just copied from the much earlier test, and I didn't update more than necessary.